### PR TITLE
Backchannel updates

### PR DESCRIPTION
**Pull request by AleksandarFilipov**
Added backchannel test where signing can be done off-socket.

if backchannel is present that used, here: 
https://github.com/aeternity/ae-channel-service/compare/backchannel-updates?expand=1#diff-835341fcd3816d33d40b79db245d9db5R865

testcases added.

### CODE REVIEW

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 1]
```diff
@@ -8,15 +8,15 @@ defmodule ClientRunner do
 
   def start_link(
         {_pub_key, _priv_key, %SocketConnector.WsConnection{}, _ae_url, _network_id, _role, _jobs,
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/client_runner.ex:10]: Such large tuples as arguments are a sign of a bad API design. But I believe I commented on that before :-)

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 2]
```diff
@@ -38,11 +38,10 @@ defmodule ClientRunner do
 
               auto_approval
             end,
-            channels_update: fn round_initiator, nonce ->
+            channels_update: fn round_initiator, nonce, method ->
               Logger.debug(
                 "callback received round is: #{inspect(nonce)} round_initiator is: #{
-                  inspect(round_initiator)
-                }}",
+                  inspect(round_initiator)} method is #{inspect method}}",
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/client_runner.ex]: With or without `()` I don't care too much, just be consistent.

File: apps/ae_socket_connector/lib/socket_connector.ex [hunk 3]
```diff
@@ -218,6 +202,11 @@ defmodule SocketConnector do
     WebSockex.cast(pid, {:transfer, amount})
   end
 
+  @spec initiate_transfer(pid, integer) :: :ok
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/socket_connector.ex]: spec is outdated

**Review by tolbrino — APPROVED** [review]

File: apps/ae_socket_connector/lib/socket_connector.ex (hunk 3 above)
Comment (AleksandarFilipov) [reply apps/ae_socket_connector/lib/socket_connector.ex]: fixed

File: apps/ae_socket_connector/lib/client_runner.ex (hunk 1 above)
Comment (AleksandarFilipov) [reply apps/ae_socket_connector/lib/client_runner.ex:10]: Yep so you did; this must be fixed (for later)

File: apps/ae_socket_connector/lib/client_runner.ex (hunk 2 above)
Comment (AleksandarFilipov) [reply apps/ae_socket_connector/lib/client_runner.ex]: fixed

**Pull request merged**